File: 187-repeated-dna-sequences/repeated-dna-sequences.js
var findRepeatedDnaSequences = function(s) {
    let output = []
    if (s.length < 10) return output
    let lookup = {}
    let j = 0

    while (j < s.length - 9) {
        let substr = s.slice(j, j + 10)
        
        if (!(substr in lookup)) lookup[substr] = 0
        lookup[substr] += 1
        if (lookup[substr] == 2) output.push(substr) 
        j += 1
    }
    return output


};

File: 187-repeated-dna-sequences/repeated-dna-sequences.py
class Solution:
    def findRepeatedDnaSequences(self, s: str) -> List[str]:
        i = 0
        lookup = {}
        output = []

        while (i < len(s) - 9):
            substring = s[i: i + 10]

            if (substring not in lookup):
                lookup[substring] = 0

            lookup[substring] += 1
            if (lookup[substring] == 2):
                output.append(substring)
            i += 1
        

        return output